var/spack/repos/builtin/packages/duckdb/package.py

```diff
@@ -16,8 +16,106 @@ class Duckdb(CMakePackage):
+    variant("parquet", default=True, description="Include parquent support in build")
+
+    # APIs
+    variant("jdbc", default=False, description="Build JDBC driver (may not work)")
+    variant("odbc", default=False, description="Build with ODBC driver (may not work)")
+    variant("python", default=False, description="Build with Python driver (may not work)")
+
+    def edit(self, spec, prefix):
+        if "+ninjabuild" in spec:
+            env["GEN"] = "ninja"
+        variant_flags = [
+            "autocomplete",
+            "cli",
+            "excel",
+            "fts",
+            "httpfs",
+            "icu",
+            "inet",
+            "jdbc",
+            "json",
+            "odbc",
+            "openssl",
+            "parquet",
+            "python",
+        ]
+        for flag in variant_flags:
+            make_flag = "BUILD_" + flag.upper()
+            if "+" + flag in spec:
+                env[make_flag] = "1"
+            elif "~" + flag in spec:
+                env[make_flag] = "0"
+
+    def url_for_version(self, version):
+        return "https://github.com/duckdb/duckdb/archive/refs/tags/v{0}.tar.gz".format(version)
+
+    def patch(self):
+        # DuckDB pulls its version from a git tag, which it can't find in the tarball
+        # and thus defaults to something arbitrary and breaks extensions.
+        # We use the Spack version to inject it in the right place during the build
+
+        v = self.spec.version
+        if v < Version("0.10.0"):
+            # Prior to version 0.10.0, this was sufficient
+            filter_file(
+                r'(message\(STATUS "git hash \$\{GIT_COMMIT_HASH\}, '
+                r'version \$\{DUCKDB_VERSION\}"\))',
+                'set(DUCKDB_VERSION "v{0}")\n\\1'.format(self.spec.version),
+                "CMakeLists.txt",
+            )
+        else:
+            # Override the fallback values that are set when GIT_COMMIT_HASH doesn't work
+            for i, n in enumerate(["MAJOR", "MINOR", "PATCH"]):
+                filter_file(
+                    r"set\(DUCKDB_{0}_VERSION \d+\)".format(n),
+                    "set(DUCKDB_{0}_VERSION {1})".format(n, v[i]),
+                    "CMakeLists.txt",
+                )
+            # Need to manually set DUCKDB_NORMALIZED_VERSION for helper scripts
+            filter_file(
+                r'(message\(STATUS "git hash \$\{GIT_COMMIT_HASH\},'
+                r" version \$\{DUCKDB_VERSION\},"
+                r' extension folder \$\{DUCKDB_NORMALIZED_VERSION\}"\))',
+                'set(DUCKDB_VERSION "v${DUCKDB_MAJOR_VERSION}'
+                '.${DUCKDB_MINOR_VERSION}.${DUCKDB_PATCH_VERSION}")'
+                '\nset(DUCKDB_NORMALIZED_VERSION "${DUCKDB_VERSION}")'
+                "\n\\1",
+                "CMakeLists.txt",
+            )
+
+    def install(self, spec, prefix):
+        mkdir(prefix.bin)
+        install("build/release/duckdb", prefix.bin)
```

Comment:
Kind of sad that the `Makefile` doesn't have an install target.

Reply:
This may become increasingly sad as the build process seems to be expanding to create more artifacts than just the binary. Might be worth addressing with them upstream.